ferveo/src/refresh.rs

```diff
@@ -0,0 +1,384 @@
+use std::{ops::Mul, usize};
+
+use ark_ec::{pairing::Pairing, AffineRepr, CurveGroup};
+use ark_ff::Zero;
+use ark_poly::{univariate::DensePolynomial, DenseUVPolynomial, Polynomial};
+use group_threshold_cryptography as tpke;
+use itertools::zip_eq;
+use rand_core::RngCore;
+use tpke::{lagrange_basis_at, PrivateKeyShare};
+
+// SHARE UPDATE FUNCTIONS:
```

Comment:
We could always put those functions into a module if we think it would help organise them better:

```
pub mod share_update {

	pub fn prepare_share_updates_for_recover ...

	pub fn apply_updates_to_private_share ...

}

// etc.

```

But on second thought, these sections will by different by the time we're done with the refactoring.

Anyway, just throwing it out there - It's an option.

Reply:
oh that's a nice idea, I didn't know that. Let's consider that for the next refactor.